### PR TITLE
Add code to handle cdc usb devices

### PR DESCRIPTION
**Pull request by tectrolabs**

### CODE REVIEW

**Comment by tectrolabs** [comment]
Made changes to support new SwiftRNG CDC class devices

File: windows-x64/SwiftRNG/USBComPort.cpp [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+*/
+
+/*+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Copyright (C) 2014-2020 TectroLabs, https://tectrolabs.com
+
+THIS SOFTWARE IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED,
+INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+This class implements the access to a SwiftRNG device over CDC USB interface on Windows platform.
```
Comment (ebelinski) [inline windows-x64/SwiftRNG/USBComPort.cpp]:
```suggestion
This class implements access to a SwiftRNG device over CDC USB interface on Windows platform.
```

File: windows-x64/SwiftRNG/USBComPort.cpp [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+/**
+* Receive data from SwiftRNG device
+*
+* @param unsigned char *rcv - a pointer to receive buffer
+* @param int sizeRcv - how many bytes to receive
+* @return 0 - successful operation, otherwise the error code
+*
+*/
+int USBComPort::receiveDeviceData(unsigned char *rcv, int sizeRcv, int *bytesReveived) {
+	DWORD actualButesReceived;
```
Comment (ebelinski) [inline windows-x64/SwiftRNG/USBComPort.cpp]: Typo:

```suggestion
	DWORD actualBytesReceived;
```

File: windows-x64/SwiftRNG/USBComPort.cpp [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+#include "USBComPort.h"
+
+USBComPort::USBComPort()
+{
+	this->deviceConnected = false;
+	this->lastError = new char[512];
+	clearErrMsg();
+}
+
+void USBComPort::initlalize() {
```
Comment (ebelinski) [inline windows-x64/SwiftRNG/USBComPort.cpp]: Typo:
```suggestion
void USBComPort::initialize() {
```

File: windows-x64/SwiftRNG/swrngapi.cpp [hunk 4]
```diff
@@ -332,20 +373,27 @@ static int swrng_snd_rcv_usb_data(SwrngContext *ctxt, char *snd, int sizeSnd, ch
  * Open SwiftRNG USB specific device.
  *
  * @param ctxt - pointer to SwrngContext structure
- * @param int deviceNum - device number (0 - for the first device or when there is only one device connected)
+ * @param int devNum - device number (0 - for the first device or when there is only one device connected)
  * @return int 0 - when open successfully or error code
  */
-int swrngOpen(SwrngContext *ctxt, int deviceNum) {
-	int status;
+int swrngOpen(SwrngContext *ctxt, int devNum) {
```
Comment (ebelinski) [inline windows-x64/SwiftRNG/swrngapi.cpp:379]:
```suggestion
int swrngOpen(SwrngContext* ctxt, int devNum) {
```

**Review by ebelinski — CHANGES_REQUESTED** [review]
I found a few typos here and there. Looks good overall, although I had a hard time figuring out the nitty gritty of what some of the code actually does, since I'm not very familiar with the USB APIs here.

**Pull request merged**